lib/account.js

```diff
@@ -1,12 +1,20 @@
 module.exports = account
 
 var addTokenToUserDoc = require('./utils/add-token-to-user-doc')
+var errors = require('./utils/errors')
 var findUserDoc = require('./utils/find-user-doc-by-username-or-id-or-token')
 
 function account (setupPromise, state, findAccountOptions) {
   return {
     tokens: {
       add: function (tokenOptions) {
+        var validPattern = /^([a-z])([a-z0-9\-_])*$/
+
+        if (typeof tokenOptions.type !== 'string' ||
+        !validPattern.test(tokenOptions.type)) {
+          return Promise.reject(errors.TOKEN_TYPE_INVALID)
+        }
```

Comment:
can you move this code into a separate file like `/lib/utils/is-valid-token-type.js` and then do `if (!isValidTokenType(tokenOptions.type)) {`?